torch/_dynamo/guards.py

```diff
@@ -1208,26 +1822,34 @@ def add_code_part(code_part, guard, log_only=False):
             **SYMPY_INTERP,
             **CLOSURE_VARS,
         }
+        globals_for_guard_fn = {"G": builder.scope["G"]}
 
-        unique_code_parts = list(unique(code_parts))
-        make_guard_fn_args = ", ".join(closure_vars.keys())
-        guard_body, pycode = build_guard_function(unique_code_parts, make_guard_fn_args)
+        if config.enable_cpp_guard_manager:
+            # Guard manager construction is complete
+            assert self.guard_manager  # to make mypy happy
+            guard_fn = self.guard_manager
```

Comment:
Can we implement calling this directly from C++ then do some perf experiments?

Reply:
Moved the call to C++ already in a previous PR. Roughly 1.5x - 1.8x guard overhead reduction on various models.